gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -23,6 +23,37 @@ from .transports.base import {{ service.name }}Transport
 from .transports.grpc import {{ service.name }}GrpcTransport
 
 
+def _get_default_mtls_endpoint(api_endpoint):
+    """Convert api endpoint to mTLS endpoint.
+    Convert "*.sandbox.googleapis.com" and "*.googleapis.com" to
+    "*.mtls.sandbox.googleapis.com" and "*.mtls.googleapis.com" respectively.
+    Args:
+        api_endpoint (Optional[str]): the api endpoint to convert.
+    Returns:
+        str: converted mTLS api endpoint.
+    """
+    if (
+        api_endpoint is None
+        or api_endpoint.find("mtls.sandbox.googleapis.com") != -1
+        or api_endpoint.find("mtls.googleapis.com") != -1
+        or api_endpoint.find(".googleapis.com") == -1
+    ):
+        # If the endpoint is already mTLS or the endpoint is not a googleapi,
+        # there is no need to convert.
+        return api_endpoint
+
+    if api_endpoint.find(".sandbox.googleapis.com") != -1:
+        return api_endpoint.replace(
+            ".sandbox.googleapis.com", ".mtls.sandbox.googleapis.com"
+        )
+
+    return api_endpoint.replace(".googleapis.com", ".mtls.googleapis.com")
+
+
+_DEFAULT_ENDPOINT = {% if service.host %}'{{ service.host }}'{% else %}None{% endif %}
+_DEFAULT_MTLS_ENDPOINT = _get_default_mtls_endpoint(_DEFAULT_ENDPOINT)
```

Comment:
I think it would be better to turn these into class attributes for the client. If they live right above `DEFAULT_OPTIONS` then there is no problem with referencing the default endpoint.

Reply:
made them class attributes